tools/devops/device-tests/templates/device-tests.yml

```
@@ -251,9 +251,14 @@ steps:
     cd $WORKING_DIR/xamarin-macios
     if [[ "$USE_XAMARIN_STORAGE" == "True" ]]; then
       if nc -z xamarin-storage 22 2>/dev/null; then
-        ssh builder@xamarin-storage "mkdir -p /volume1/storage/$XAMARIN_STORAGE_PATH"
-        echo '##vso[task.setvariable variable=XAMARIN_STORAGE_FAILED;isOutput=true]false'
-        echo '##vso[task.setvariable variable=XAMARIN_STORAGE_PATH;isOutput=true]$XAMARIN_STORAGE_PATH'
+        EC=0
+        MKSTORAGE="ssh builder@xamarin-storage "mkdir -p /volume1/storage/$XAMARIN_STORAGE_PATH""
+        eval $MKSTORAGE || EC=$?
+        if [ $EC -eq 0 ]; then
+          echo '##vso[task.setvariable variable=XAMARIN_STORAGE_PATH;isOutput=true]$XAMARIN_STORAGE_PATH'
+        else
+          echo '##vso[task.setvariable variable=XAMARIN_STORAGE_FAILED;isOutput=true]false'
```

This is the otherway around, we did not get a 0, ergo we could not set the path, so we did fail
```suggestion
          echo '##vso[task.setvariable variable=XAMARIN_STORAGE_FAILED;isOutput=true]true'
```